jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildDockerTask.java

```diff
@@ -87,7 +90,34 @@ public void dockerClient(Action<? super DockerClientParameters> action) {
   public void buildDocker()
       throws IOException, BuildStepsExecutionException, CacheDirectoryCreationException,
           MainClassInferenceException {
-    Path dockerExecutable = dockerClientParameters.getExecutablePath();
+    Preconditions.checkNotNull(jibExtension);
+    Path dockerExecutable = jibExtension.getDockerClient().getExecutablePath();
+    Map<String, String> dockerEnvironment = jibExtension.getDockerClient().getEnvironment();
+
+    // Check deprecated parameters
+    if (getDockerClient().getExecutable() != null) {
+      if (dockerExecutable != null) {
+        throw new GradleException(
+            "Cannot configure 'jibDockerBuild.dockerClient.executable' and 'jib.dockerClient.executable' simultaneously");
+      }
+      getProject()
+          .getLogger()
+          .warn(
+              "'jibDockerBuild.dockerClient.executable' is deprecated; use 'jib.dockerClient.executable' instead.");
+      dockerExecutable = getDockerClient().getExecutablePath();
```

Comment:
Why overwrite when `dockerExecutable` is already configured?

Maybe we can simplify the code a bit:
```java
if (getDockerClient().getExecutable() != null || !getDockerClient().getEnvironment().isEmpty()) {
  ...warn(...);
}
if ((dockerExecutable != null && getDockerClient().getExecutable() != null) || (!dockerEnvironment.isEmpty() && !getDockerClient().getEnvironment().isEmpty()) {
    throw new GradleException("Cannot configure 'jib.dockerClient' and 'jibDockerBuild.dockerClient' ...");
}

Reply:
Ah I jumped on this too soon. The reason for the overwrite is because `PluginConfigurationProcessor` needs to use either `getDockerClient().getExecutable()`, or `jibExtension.getDockerClient().getExecutable()`. If we don't do the overwrite, then this will break users since their original configuration won't be passed forward.